drivers/adc-dac/ad74416h/ad74416h.c

```diff
@@ -530,13 +530,34 @@ int ad74416h_set_channel_function(struct ad74416h_desc *desc,
 {
 	int ret;
 
+	ret = ad74416h_reg_update(desc, AD74416H_CH_FUNC_SETUP(ch),
+				  AD74416H_CH_FUNC_SETUP_MSK, AD74416H_HIGH_Z);
+	if (ret)
+		return ret;
+
+	if (desc->channel_configs[ch].vout_range == AD74416H_VOUT_RANGE_0_12V)
+		ret = ad74416h_reg_update(desc, AD74416H_DAC_CODE(ch),
+					  AD74416H_DAC_CODE_MSK, 0x0);
+	else
+		ret = ad74416h_reg_update(desc, AD74416H_DAC_CODE(ch),
+					  AD74416H_DAC_CODE_MSK, 0x8000);
```

Comment:
A macro would be nicer.

Reply:
actually I will call `ad74416h_dac_voltage_to_code` here which handles also the dac range :)